apps/api/R/runs.R

```diff
@@ -102,6 +106,102 @@ getRunDetails <- function(id, res){
     return(list(error="Run with specified ID was not found"))
   }
   else {
-    return(qry_res)
+    # Convert the response from tibble to list
+    response <- list()
+    for(colname in colnames(qry_res)){
+      response[colname] <- qry_res[colname]
+    }
+    
+    # If outputs exist on the host, add them to the response
+    outdir <- paste0(Sys.getenv("DATA_DIR", "/data/"), "workflows/PEcAn_", response$workflow_id, "/out/", run_id)
+    if(dir.exists(outdir)){
+      response$outputs <- getRunOutputs(outdir)
+    }
+    
+    return(response)
+  }
+}
+
+#################################################################################################
+
+#' Plot the results obtained from a run
+#' @param run_id Run id (character)
+#' @param year the year this data is for
+#' @param yvar the variable to plot along the y-axis.
+#' @param xvar the variable to plot along the x-axis, by default time is used.
+#' @param width the width of the image generated, default is 800 pixels.
+#' @param height the height of the image generated, default is 600 pixels.
+#' @return List of runs (belonging to a particuar workflow)
+#' @author Tezan Sahu
+#* @get /<run_id>/graph/<year>/<y_var>
+#* @serializer contentType list(type='image/png')
+
+plotResults <- function(run_id, year, y_var, x_var="time", width=800, height=600, res){
```

Comment:
Not that there's anything 'wrong' with this, but I'd say that grabbing/downloading the model output itself is going to be FAR more important to most users than downloading an image file. Probably grabbing the whole netCDF would be the best place to start, but I could see a lot of folk wanting to subset variables, locations, data ranges, and ensemble members. The latter support it built into THREDDS, but (1) not all of our PEcAn servers support THREDDS (but maybe the right thing to do is to fix that) and (2) regardless we should have some helper functions in the R package to help people download output rather than having to learn to work with THREDDS directly (which is nontrivial)

I think there's also a REALLY high need, even if it's just internal to the project, to be able to correctly grab INPUTS and POSTERIORS via the API. The former will be pretty important to a lot of enduser analyses as well (e.g. plot predicted fluxes vs meteorological inputs)

Reply:
This function is needed in the current web application. The current web application can not create the graphs we had in the VM version. This will allow me/us to call this endpoint to get an image back.

Reply:
As for the ability to download the outputs/inputs/posteriors I agree, we just have to find the right functions, but I think we have a lot of this framework in place.

Reply:
Ahh, yes that makes perfect sense.
As we discussed during last week's planning meeting, we should think about whether it makes sense to rework the whole PHP interface to use the API rather than it's current approach of direct execution of R code.

Reply:
shhhhh, but that is my plan :) and maybe even use R and leaflet or other things.